src/macaron/dependency_analyzer/cyclonedx_mvn.py

```diff
@@ -63,24 +69,19 @@ def collect_dependencies(self, dir_path: str) -> dict[str, DependencyInfo]:
             A dictionary where artifacts are grouped based on "artifactId:groupId".
         """
         # Load the top level file separately as it has different content.
-        top_path = Path(os.path.join(dir_path, "target", self.file_name))
-        top_path_altered = False
-        if not os.path.exists(top_path):
-            # If the expected bom file does not exist, allow other named .json files instead.
-            possible_paths = glob.glob(os.path.join(dir_path, "target", "*.json"))
-            if possible_paths:
-                if len(possible_paths) > 1:
-                    logger.debug("Too many JSON SBOM files found. Expected: 1, Found: %s", len(possible_paths))
-                    return {}
-                top_path = Path(possible_paths[0])
-                top_path_altered = True
+        possible_paths = glob.glob(os.path.join(dir_path, "target", "*.json"))
```

Comment:
I preferred the previous version to first check for the expected file name. If that file cannot be found, we can look for other JSON files. This can avoid missing the BOM files when a project uses expected names but also produces other JSON files.